### ISSUE TITLE
Pressing back button on new tab closes tab

### ISSUE THREAD

**Issue by npark-mozilla** [mozilla-mobile/focus-android#1678]
I was wondering whether the behavior I'm seeing matches expectations - When doing the following,
1. Go to a webpage A
2. Open a new tab to open webpage B.

3. Go back to webpage A, browse a few more sites, and press back button until there is no sites to go back to, then press back button again
Expected: ?
Actual: closes the new tab, remains in webpage A

3. From the new tab, press back button.
Expected: ?
Actual: closes the new tab, returns to webpage A

Not sure whether it's a bug, but I didn't expect new tabs would be closing while I was pressing back button, as I didn't know that I reached the initial page that I visited. @antlam thoughts?

**Comment by liuche** [comment]
Yeah, these problems are popping up more: see #1569 - there are a few more that popped up in triage but I can't remember what they were. If I see them again I will link them to #1569.

**Comment by pocmo** [comment]
Yes, this is the intended behavior for now (and btw. in Fennec too): You will leave the app when pressing back in tabs that are opened from external apps.

**Comment by bbinto** [comment]
+ @lime124 to review

**Comment by antlam** [comment]
Sorry for the late reply @npark-mozilla !

For context, (and in short) yes to what @pocmo said. this is intended to be platform agnostic UX.
